recipes/nifti_clib/all/conanfile.py

```diff
@@ -0,0 +1,160 @@
+import os
+
+
+required_conan_version = ">=1.53.0"
+
+
+class NiftiClibConan(ConanFile):
+    name = "nifti_clib"
+    description = "C libraries for NIFTI support"
+    license = "LicenseRef-LICENSE"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/NIFTI-Imaging/nifti_clib"
+    topics = ("image")
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "use_nifti2": [True, False],
+        "use_cifti": [True, False],
+        "use_fslio": [True, False]
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "use_nifti2": True,
+        "use_cifti": False,  # seems to be beta?
+        "use_fslio": False  # Note in CMakeLists.txt: "If OFF, The copyright of this code is questionable for inclusion with nifti."
+    }
+
+    def export_sources(self):
+        export_conandata_patches(self)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+        self.settings.rm_safe("compiler.cppstd")
+        self.settings.rm_safe("compiler.libcxx")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        self.requires("zlib/1.3.1")
+        if self.options.use_cifti:
+            self.requires("expat/2.6.2")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        tc.variables["NIFTI_INSTALL_NO_DOCS"] = True
+        tc.variables["USE_NIFTI2_CODE"] = self.options.use_nifti2
+        tc.variables["USE_CIFTI_CODE"] = self.options.use_cifti
+        tc.variables["USE_FSL_CODE"] = self.options.use_fslio
+        tc.variables["NIFTI_BUILD_TESTING"] = False  # maybe this should be false? It downloads extra test data
+        if is_msvc(self):
+            tc.variables["USE_MSVC_RUNTIME_LIBRARY_DLL"] = not is_msvc_static_runtime(self)
+            tc.variables["CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS"] = True
+            tc.preprocessor_definitions["_CRT_SECURE_NO_WARNINGS"] = 1
+        tc.generate()
+        tc = CMakeDeps(self)
+        tc.generate()
+        tc = VirtualBuildEnv(self)
+        tc.generate(scope="build")
+
+    def _patch_sources(self):
+        apply_conandata_patches(self)
+
+    def build(self):
+        self._patch_sources()
+
+        if is_msvc(self):  # patch in __declspec(dllimport) for a global variable that is used
+            to_replace = "extern char const * const inam[];"
+            fixed = "__declspec(dllimport) " + to_replace
+            replace_in_file(self, os.path.join(self.source_folder, "nifticdf", "nifticdf.h"), to_replace, fixed, strict=True)
```

Comment:
This is exactly the kind of changes we're trying to avoid - releated to my previous msvc comments (Either way, thanks for taking the time to try to "fix it", we appreciate it nontheless)